### PR TITLE
🎨 Move Brig mappings to a common class for Paper and Commodore

### PR DESCRIPTION
**Pull request by jmanpenilla**
This doesn't break anything on Paper, but in my testing of this it didn't work for Commodore. On top of that it seems like Commodore is kind of borked in general. Valid commands get shown as invalid and suggestions are messed up when commodore is active (with or without the changes in this PR)

### CODE REVIEW

**Comment by Sauilitired** [comment]
This needs to be tested with the new commodore changes. That should hopefully fix the commodore related issues. If there are still issues, this can still be merged but we need to note which issues that are still around so they can be fixed separately.

So essentially, test and open issues if necessary. When done, this can be merged :+1:

File: cloud-minecraft/cloud-bukkit/src/main/java/cloud/commandframework/bukkit/BukkitBrigadierMapper.java [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+        return Class.forName(String.format("net.minecraft.server.%s.Argument%s", this.nmsVersion, argument));
+    }
+
+    /**
+     * Attempt to register a mapping between a type and a NMS argument type
+     *
+     * @param type        Type to map
+     * @param constructor Constructor that construct the NMS argument type
+     */
+    public void mapSimpleNMS(@NonNull final Class<?> type,
```
Comment (Sauilitired) [inline cloud-minecraft/cloud-bukkit/src/main/java/cloud/commandframework/bukkit/BukkitBrigadierMapper.java:131]:
```suggestion
    public void mapSimpleNMS(final @NonNull Class<?> type,
```

File: cloud-minecraft/cloud-bukkit/src/main/java/cloud/commandframework/bukkit/BukkitBrigadierMapper.java [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+                             @NonNull final Constructor<?> constructor) {
```
Comment (Sauilitired) [inline cloud-minecraft/cloud-bukkit/src/main/java/cloud/commandframework/bukkit/BukkitBrigadierMapper.java:132]:
```suggestion
                             final @NonNull Constructor<?> constructor) {
```

File: cloud-minecraft/cloud-bukkit/src/main/java/cloud/commandframework/bukkit/BukkitBrigadierMapper.java [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+
+    /**
+     * Attempt to retrieve an NMS argument type
+     *
+     * @param argument Argument type name
+     * @return Argument class
+     * @throws Exception If the type cannot be retrieved
+     */
+    @NonNull
+    private Class<?> getNMSArgument(@NonNull final String argument) throws Exception {
```
Comment (Sauilitired) [inline cloud-minecraft/cloud-bukkit/src/main/java/cloud/commandframework/bukkit/BukkitBrigadierMapper.java:121]:
```suggestion
    private @NonNull Class<?> getNMSArgument(final @NonNull String argument) throws Exception {
```

File: cloud-minecraft/cloud-bukkit/src/main/java/cloud/commandframework/bukkit/BukkitBrigadierMapper.java [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
+        }
+    }
+
+    /**
+     * Attempt to register a mapping between a type and a NMS argument type
+     *
+     * @param type                 Type to map
+     * @param argumentTypeSupplier Supplier of the NMS argument type
+     */
+    public void mapComplexNMS(@NonNull final Class<?> type,
```
Comment (Sauilitired) [inline cloud-minecraft/cloud-bukkit/src/main/java/cloud/commandframework/bukkit/BukkitBrigadierMapper.java:156]:
```suggestion
    public void mapComplexNMS(final @NonNull Class<?> type,
```

File: cloud-minecraft/cloud-bukkit/src/main/java/cloud/commandframework/bukkit/BukkitBrigadierMapper.java [hunk 5]
```diff
@@ -0,0 +1,167 @@
...
+                              @NonNull final Supplier<ArgumentType<?>> argumentTypeSupplier) {
```
Comment (Sauilitired) [inline cloud-minecraft/cloud-bukkit/src/main/java/cloud/commandframework/bukkit/BukkitBrigadierMapper.java:157]:
```suggestion
                              final @NonNull Supplier<ArgumentType<?>> argumentTypeSupplier) {
```

File: cloud-minecraft/cloud-bukkit/src/main/java/cloud/commandframework/bukkit/BukkitBrigadierMapper.java [hunk 6]
```diff
@@ -0,0 +1,167 @@
...
+import org.checkerframework.checker.nullness.qual.NonNull;
+
+import java.lang.reflect.Constructor;
+import java.lang.reflect.InvocationTargetException;
+import java.util.UUID;
+import java.util.function.Supplier;
+import java.util.logging.Level;
+
+@SuppressWarnings("ALL")
+public final class BukkitBrigadierMapper<C> {
```
Comment (Sauilitired) [inline cloud-minecraft/cloud-bukkit/src/main/java/cloud/commandframework/bukkit/BukkitBrigadierMapper.java:45]: Might be good to document this?

**Review by Sauilitired — CHANGES_REQUESTED** [review]

**Comment by jmanpenilla** [comment]
Commodore issues seem fixed with the new changes. Updated the PR with your suggestions. I found out that Commodore wasn't mapping namespaced commands but fixed it in #35

**Comment by Sauilitired** [comment]
Alright then we should be good to. I'm going to do more testing later on but I want to get these PRs merged sooner rather than later to prevent more merge confllicts :p

**Pull request merged**